### PR TITLE
Comments and fixes

### PR DESCRIPTION
**Pull request by jenrik**

### CODE REVIEW

**Comment by jenrik** [comment]
[Review PR "Comments and fixes"](https://trello.com/c/B8Y390ut/63-review-pr-comments-and-fixes)

**Review by NicEastvillage — APPROVED** [review]
Nice

**Review by cogitantium — APPROVED** [review]
Nice

File: src/com.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ pub trait Broker<V: Hash + Eq + PartialEq + Clone> {
     /// Send message to worker with id `to`
     fn send(&self, to: WorkerId, msg: Message<V>);
 
-    /// Signal to all workers a terminate because a result have been
+    /// Signal to all workers to terminate because a result have been
```
Comment (MathiasMehl) [inline src/com.rs:10]:
```suggestion
    /// Signal to all workers to terminate because a result has been found
```

File: src/com.rs [hunk 2]
```diff
@@ -58,6 +59,8 @@ impl<V: Hash + Eq + PartialEq + Clone> ChannelBroker<V> {
             msg_receivers.push(receiver);
         }
 
+        // Create a termination channel foreach
```
Comment (MathiasMehl) [inline src/com.rs:62]:
```suggestion
        // Create a termination channel foreach worker
```

File: src/com.rs [hunk 3]
```diff
@@ -58,6 +59,8 @@ impl<V: Hash + Eq + PartialEq + Clone> ChannelBroker<V> {
...
+        // These are used to signal early termination because the final assigment of `v0` have been discovered
```
Comment (MathiasMehl) [inline src/com.rs:63]:
```suggestion
        // These are used to signal early termination because the final assignment of `v0` has been discovered
```

**Review by GeneralDenmark — APPROVED** [review]

File: src/common.rs [hunk 4]
```diff
@@ -28,14 +28,19 @@ pub enum Edges<V: Hash + Eq + PartialEq + Clone> {
...
 }
 
+/// Inter-Worker communication
 #[derive(Clone)]
 pub enum Message<V: Hash + Eq + PartialEq + Clone> {
+    /// Send when a hyper-edge is discovered
     HYPER(HyperEdge<V>),
+    /// Send when a negation-edge is discovered
     NEGATION(NegationEdge<V>),
+    /// Send from a worker that need the final assignment of `vertex` but is not the owner of the vertex.
```
Comment (MathiasMehl) [inline src/common.rs:38]:
```suggestion
    /// Send from a worker that needs the final assignment of `vertex` but is not the owner of the vertex.
```

File: src/common.rs [hunk 5]
```diff
@@ -28,14 +28,19 @@ pub enum Edges<V: Hash + Eq + PartialEq + Clone> {
...
     REQUEST {
         vertex: V,
         worker_id: WorkerId,
     },
+    /// Send from the owner `vertex` to all workers that have request the final assignment of `vertex`
```
Comment (MathiasMehl) [inline src/common.rs:43]:
```suggestion
    /// Send from the owner of `vertex` to all workers that have requested the final assignment of `vertex`
```

File: src/edg.rs [hunk 6]
```diff
@@ -62,28 +65,36 @@ pub fn distributed_certain_zero<
 #[derive(Debug)]
 struct Worker<B: Broker<V> + Debug, G: ExtendedDependencyGraph<V>, V: Vertex> {
     id: WorkerId,
+    /// Number of workers working on solving the query. This is used as part of the static allocation scheme, see `crate::Worker::vertex_owner`.
     worker_count: u64,
     v0: V,
     assignment: HashMap<V, VertexAssignment>,
     depends: HashMap<V, HashSet<Edges<V>>>,
+    /// Map of workers that needs to be send a message once the final assigment of a vertex is known.
```
Comment (MathiasMehl) [inline src/edg.rs:73]:
```suggestion
    /// Map of workers that need to be sent a message once the final assignment of a vertex is known.
```

**Review by MathiasMehl — COMMENTED** [review]
Minor grammar fixes

**Pull request merged**